src/util/devices.tsx

```diff
@@ -98,3 +111,9 @@ export const getExistingDeviceNames = (
 
   return existingDeviceNames;
 };
+
+export const isVolume = (
```

Comment:
What is the difference between isVolume and isVolumeDevice? Can they be combined to a single function?

Reply:
slightly tricky from a typescript perspective, this function operates on `LxdStorageVolume` which is very different to the device types that `isVolumeDevice` works with

Reply:
Looking at the two usages of this function, I think we can simplify, leading to a removal of the function.

1. the `changeDiskDevice` function can only ever receive a `LxdStorageVolume` as the `diskDevice`. So we should remove the ambiguous other possible type of `LydDiskDevice` for the `diskDevice`. Then rename the parameter there from `diskDevice` to `volume` and we don't need to call `isVolume` inside.

2.  the `addDiskDevice` function should not receive a parameter `device: LxdStorageVolume | LxdDiskDevice`, because a volume is not a device. I think in `AttachDiskDeviceBtn` we should change the prop to `setValue: (device: LxdDiskDevice) => void`. Inside the `AttachDiskDeviceBtn::handleFinish` we can do the translation from volume to `LxdDiskDevice`. If we fill the parameters correctly, we have no branching any more in `addDiskDevice`. This removes the ambiguity and the second use case of `isVolume`.

Then we can remove `isVolume`.

Reply:
For 2. the translation needs to happen in `AttachDiskDeviceModal`, one more layer downwards, where we receive the volume from `CustomVolumeModal`. Only there it is not ambiguous. In the `AttachDiskDeviceBtn` we would also have to differentiate between disk and volume.

Reply:
Thanks this is great, regarding 1. I must have been out of it, not sure how it ended up like that!

Reply:
One note, I've made the change but moved the translation down into `CustomVolumeModal`. That way, both `HostPathDeviceModal` and `CustomVolumeModal` will pass a `LxdDiskDevice` back up to the `DiskDeviceFormCustom` component for both adding and editing disk devices regardless of the disk type. Let me know what you think.

Reply:
I am not sure raising it to the CustomVolumeModal is the right choice. Maybe that component should not deal with the concept of devices. This makes the change device function a bit complicated to gasp. Maybe keep on drilling the volume down to the change device function, this might make it more obvious? Or there coudl be a way to make the change function easier to understand.

Reply:
I've moved the translation into `AttachDiskDeviceModal` and changed the function in `DiskDeviceFormCustom` to `changeVolume`.

Reply:
Thanks for the changes! I'll have a closer look at this shortly. To answer your question: That was a continuation of the thought process of the implications of having the translation in the choose custom volume component. It should be resolved with your recent changes.

Reply:
Actually no, only in the AttachDiskDeviceModal we have two branches where we receive the volume or the host path as a device object. My thought was that there is a natural distinction in the modal as it is receiving the two different types in different callbacks for the different parents.

In the button component we would have condensed that I to a single callback. Hope this makes sense.

Reply:
Still recent changes should be all right.

Reply:
Thanks for the quick feedback :) The only thing that I wasn't sure about is the comment regarding the button component. With that I think all is in order, this comment can be resolved if you are happy with the changes